ethereumj-core/src/main/java/org/ethereum/crypto/ECKey.java

```diff
@@ -597,14 +885,18 @@ public static ECKey signatureToKey(byte[] messageHash, String signatureBase64) t
      */
     public byte[] decryptAES(byte[] cipher){
 
-        if (priv == null)
+        if (privKey == null) {
             throw new MissingPrivateKeyException();
+        }
+        if (!(privKey instanceof BCECPrivateKey)) {
+            throw new UnsupportedOperationException("Cannot use the private key as an AES key");
+        }
```

Comment:
This method `ECKey#decryptAES(byte[])` should really just be deleted. The only place this method is used is in a standalone test in `ECKeyTest`.
It is rather odd to take the scalar value of a EC private key and use it as a AES key.
I couldn’t deduce from the git history why this method was added in the first place.